### DISCUSSION TITLE
Problem: This account is empty

### DISCUSSION THREAD

**Discussion by igzmanuelmartinvivaldi** [devmount/third-stats#251] [Q&A] [answered]
All my Gmail's accounts and folders are listed, but 0 emails are found (v1.4.0)
This message is displayed:
"This account is empty, no emails here. "
Any idea or help?
Thanks

**Comment by devmount** [comment, marked as answer]
Hi Manuel,
which version of Thunderbird do you use? Which OS? Can you post a screenshot of the console output? (Thunderbird > add-ons > gear menu > debug add-ons > ThirdStats Inspect button)

**Reply by igzmanuelmartinvivaldi** [reply]
Thunderbird 78.7.1 (64-bit) on Windows 10
![image](https://user-images.githubusercontent.com/2234919/107947697-08983300-6f93-11eb-9933-2e8c2c2977fd.png)
Thanks for the fast reply. Hope this helps

**Reply by devmount** [reply]
If you now (with the console open) reopen the stats page, do any new error messages appear?

**Reply by igzmanuelmartinvivaldi** [reply]
![image](https://user-images.githubusercontent.com/2234919/107960457-f45d3180-6fa4-11eb-8397-a9667ae10671.png)

**Reply by devmount** [reply]
Ok, nothing of interest so far. If you now (with the console open) process the account again, do any new error messages appear?
![image](https://user-images.githubusercontent.com/5441654/107960893-79484b00-6fa5-11eb-9d6b-36dce94e4a36.png)

**Reply by igzmanuelmartinvivaldi** [reply]
Thanks, but nothing appeared after changing accounts and refreshing the data
![image](https://user-images.githubusercontent.com/2234919/107964131-430cca80-6fa9-11eb-9f8f-ad5e0decec9a.png)

**Reply by devmount** [reply]
Ok, this looks like ThirdStats is working corrently but doesn't recognize a single email. Here is a snippet to count all messages of all folders of all accounts, maybe this throws an error we can work with.

```javascript
let accounts = await messenger.accounts.list(); let allfolders = []; let count = 0;
accounts.map(a => { function traverse(folders) { if (!folders) return; for (let f of folders) { allfolders.push(f); traverse(f.subFolders); }} traverse(a.folders);})
await Promise.all(allfolders.map(async f => { let page = await messenger.messages.list(f); page.messages.map(m => count++); while (page.id) { page = await messenger.messages.continueList(page.id); page.messages.map(m => count++); } })).then(() => { console.log(count); });
```

Please paste it into the console and hit <kbd>Enter</kbd>. The output can take some time, according to the number of messages you have. Let me know, if you now have any error messages.

**Reply by igzmanuelmartinvivaldi** [reply]
Thanks for all the support , I'm sorry for my problems, looks like I'm the only one.
I ran the script and received several messages, hope this helps (errors identical to the first image repeated about 5 times):
![image](https://user-images.githubusercontent.com/2234919/108055942-ef55bc00-7050-11eb-9e01-5f13f01586ca.png)
![image](https://user-images.githubusercontent.com/2234919/108056053-16ac8900-7051-11eb-9bc4-c25e9799cc25.png)

**Reply by devmount** [reply]
You're welcome. Well, you're not the only one. Given the very last error message *Uncaught (in promise) Error: folder not found*, I afraid you're affected by #19 too. In short: Thunderbird 78 is not able to handle all types of folders properly. This was fixed in version 85. As a user of Thunderbird stable releases, you'll have to wait until the next major release, containing this fix (which will probably be version 91).

Sorry, but I can't do more for you at this point.

Final note: none of the ThirdStats users having this issue had time to test, if ThirdStats works in the [current Thunderbird beta](https://www.thunderbird.net/en-US/#channel). If you have the possibility to test it and post your results in #19, it would be a great help. 🙏🏻

**Reply by igzmanuelmartinvivaldi** [reply]
Thanks a lot Andreas, for all the help and the quick responses, it was a lot more I could expect. I didn't suspected this was the same than in issue #19 
I´ll try to test the latest beta and report.
I renamed the folder and I got a new problem with '/[IGZ] Clientes/EV_Evoluciona/+Fin' so looks like the problem happens with the '+' char (and maybe with other special chars)

**Reply by devmount** [reply]
My pleasure 😊 Thanks in advance for testing it in the latest beta! We will see if it's really related to #19 then.

> I renamed the folder and I got a new problem with '/[IGZ] Clientes/EV_Evoluciona/+Fin' so looks like the problem happens with the '+' char (and maybe with other special chars)

I'll test this and see, if I can reproduce this on my side. Thanks for the hint!

**Reply by devmount** [reply]
I tested folder names with `+` now:

TB 78.8: ❌
TB 86.0: ❌
TB 87.0: ❌

So it looks like the `+` is never working in folder names. So the only solution for you now is to rename your folders. I started [a corresponding thread](https://thunderbird.topicbox.com/groups/addons/T5dee8bb7bbf0b70a/plus-sign-in-folder-names-is-not-working) in TBs topicbox.

**Reply by devmount** [reply]
FYI: There is an open bug report for `%` sign in folder names:
https://bugzilla.mozilla.org/show_bug.cgi?id=1684327

Thanks to you, the `+` was added too. This is still unassigned, so we don't know, when this will get fixed.

**Reply by devmount** [reply]
@IGZmanuelMartinVivaldi This is now fixed on Thunderbirds end in version 90. So with the next major release of Thunderbird next month, you can use special characters in folder names again.
